cmake/Modules/LibAddHaskellPlugin.cmake

```diff
@@ -40,272 +40,311 @@ macro (add_haskell_plugin target)
-									endif (GHC_PRIM_LIB)
-								else (GHC_GMP_LIB)
-									remove_plugin (${target} "GHC_GMP_LIB not found")
-								endif (GHC_GMP_LIB)
-							else (GHC_BASE_LIB)
-								remove_plugin (${target} "GHC_BASE_LIB not found")
-							endif (GHC_BASE_LIB)
-						else (GHC_RTS_LIB)
-							remove_plugin (${target} "GHC_RTS_LIB not found")
-						endif (GHC_RTS_LIB)
-					else (HAVE_HASKELL_BINDING)
-						remove_plugin (${target} "haskell bindings are not included in the cmake configuration")
-					endif (HAVE_HASKELL_BINDING)
-				else (HASKELL_FOUND)
-					remove_plugin (${target} ${HASKELL_NOTFOUND_INFO})
-				endif (HASKELL_FOUND)
-			else (PLUGINPROCESS_FOUND)
-				remove_plugin (${target} "${PLUGINPROCESS_NOTFOUND_INFO}, but required for haskell plugins")
-			endif (PLUGINPROCESS_FOUND)
-		else (NOT ENABLE_ASAN)
-			remove_plugin (${target} "haskell plugins are not compatible with ENABLE_ASAN")
-		endif (NOT ENABLE_ASAN)
+												DEPENDS c2hs_haskell
+													${PLUGIN_SOURCE_FILES}
+													${HASKELL_ADD_SOURCES_TARGET}
+												APPEND)
+											add_custom_target (${target}
+													   ALL
+													   DEPENDS ${PLUGIN_HASKELL_NAME})
+
+											if (BUILD_SHARED OR BUILD_FULL)
+												install (
+													DIRECTORY
+														"${CMAKE_CURRENT_BINARY_DIR}/haskell"
+														DESTINATION
+														"lib${LIB_SUFFIX}/elektra/")
+											endif (BUILD_SHARED OR BUILD_FULL)
+
+											set (
+												HASKELL_RPATH
+												"${GHC_RTS_PATH}"
+												"${GHC_BASE_PATH}"
+												"${GHC_PRIM_PATH}"
+												"${GHC_GMP_PATH}"
+												"${CMAKE_INSTALL_RPATH}"
+												"${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}/elektra/haskell"
+												)
+
+										else (GHC_PRIM_LIB)
+											remove_plugin (${target} "GHC_PRIM_LIB not found")
+										endif (GHC_PRIM_LIB)
+									else (GHC_GMP_LIB)
+										remove_plugin (${target} "GHC_GMP_LIB not found")
+									endif (GHC_GMP_LIB)
+								else (GHC_BASE_LIB)
+									remove_plugin (${target} "GHC_BASE_LIB not found")
+								endif (GHC_BASE_LIB)
+							else (GHC_RTS_LIB)
+								remove_plugin (${target} "GHC_RTS_LIB not found")
+							endif (GHC_RTS_LIB)
+						else (HAVE_HASKELL_BINDING)
+							remove_plugin (${target}
+								       "haskell bindings are not included in the cmake configuration")
+						endif (HAVE_HASKELL_BINDING)
+					else (HASKELL_FOUND)
+						remove_plugin (${target} ${HASKELL_NOTFOUND_INFO})
+					endif (HASKELL_FOUND)
+				else (PLUGINPROCESS_FOUND)
+					remove_plugin (${target} "${PLUGINPROCESS_NOTFOUND_INFO}, but required for haskell plugins")
+				endif (PLUGINPROCESS_FOUND)
+			else (NOT ENABLE_ASAN)
+				remove_plugin (${target} "haskell plugins are not compatible with ENABLE_ASAN")
+			endif (NOT ENABLE_ASAN)
+		else (NOT BUILD_STATIC AND NOT BUILD_FULL)
+			remove_plugin (${target} "haskell plugins are not compatible with BUILD_STATIC or BUILD_FULL")
```

Comment:
This is too strict. It will remove your plugin *altogether* if someone has selected these build variants. What you want is to only exclude your plugin from the BUILD_STATIC AND BUILD_FULL builds.

So at the moment, if someone wants to compile Elektra in full and shared, your plugin also gets removed also in shared although it would work there.

Reply:
is it possible to only exclude it from certain builds? should i not remove it upon build/static and build/full but just don't call any functionality in that case? (e.g. a return on top)

Reply:
You need to add a "ONLY_SHARED" feature in cmake/Modules/LibAddPlugin.cmake and in this case the static/full should not be added. (In a quick look it looks like only an if (NOT ONLY_SHARED) around the last two lines is needed.)

Reply:
should it emit some message in that case?

Reply:
You need to handle the case if the plugin should be included but BUILD_SHARED is not active (maybe exclude the plugin then?). And you could update the "Include Plugin regexdispatcher" -> "Include Plugin regexdispatcher for shared builds" or similar.

Adding further new lines are most likely overlooked and rather increase confusion.